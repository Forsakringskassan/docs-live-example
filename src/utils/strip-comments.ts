/**
 * @internal
 */
export function stripComments(html: string): string {
    const commentPattern = /<!--.*?-->/g;
    return html.replace(commentPattern, "");
}
