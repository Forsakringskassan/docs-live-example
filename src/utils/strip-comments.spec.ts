import { stripComments } from "./strip-comments";

it("should remove comments", () => {
    expect.assertions(1);
    const html = /* HTML */ `
        <div class="my-class">
            <!-- I am a comment! -->
            <!--v-if-->
            <!---->
        </div>
    `;
    expect(stripComments(html)).toMatchSnapshot();
});

it("should remove comments that contain tags", () => {
    expect.assertions(1);
    const html = /* HTML */ `<!-- foo <div> bar -->`;
    expect(stripComments(html)).toBe("");
});
