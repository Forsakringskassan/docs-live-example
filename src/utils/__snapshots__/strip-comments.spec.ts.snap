// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should remove comments 1`] = `
"
        <div class="my-class">
            
            
            
        </div>
    "
`;
